components/esp_common/include/esp_err.h

```diff
@@ -92,6 +92,18 @@ void _esp_error_check_failed_without_abort(esp_err_t rc, const char *file, int l
 #endif
 /** @endcond */
 
+/**
+ * Macro that returns the error code if an error occurs, optionally running extra code.
+ * The extra code can be placed after the function thay may error and it will only run if the error occurs.
+ */
+#define ESP_RETURN_ON_ERR(x, ...) do {        \
```

Comment:
I suggest renaming it as the macros in this file start with **ESP_ERROR_CHECK_** . It could be something like ESP_ERROR_RETURN, or ESP_ERROR_CHECK_RETURN.

Additionally, it would be good to have an example of usage. Could you add in the description of the macro how to use it, something like this, I guess:

```
 * @example
 * esp_err_t get_data()
 * {
 *     ESP_ERROR_CHECK_RETURN(read_file())
 *     ESP_ERROR_CHECK_RETURN(read_file(), close_file())
 *     return ESP_OK;
 * }
```